executor/index_merge_reader.go

```diff
@@ -490,12 +490,13 @@ func (e *IndexMergeReaderExecutor) Next(ctx context.Context, req *chunk.Chunk) e
 		if resultTask == nil {
 			return nil
 		}
-		for resultTask.cursor < len(resultTask.rows) {
-			req.AppendRow(resultTask.rows[resultTask.cursor])
-			resultTask.cursor++
-			if req.NumRows() >= e.maxChunkSize {
-				return nil
-			}
+		if resultTask.cursor < len(resultTask.rows) {
+			offset := len(resultTask.rows) - resultTask.cursor
+			req.AppendRows(resultTask.rows[resultTask.cursor : resultTask.cursor+offset])
```

Comment:
Need compare `e.maxChunkSize-req.NumRows()` and `offset` here?

Reply:
the master code not comare,so i delete it .